server/models.py

```diff
@@ -11,7 +11,8 @@ class User(db.Model):
     is_active = db.Column(db.Boolean, default=True, server_default='true')
     created_at = db.Column(db.DateTime, default=db.func.current_timestamp())
     watchlist = db.relationship('UserMovies',
-                                backref='watchers')
+                                backref='watchers',
+                                cascade="all, delete-orphan")
```

Comment:
I don't fully understand this, but this stack overflow answer said it was necessary in order to delete an assoctioan object from a model's children https://stackoverflow.com/questions/23699651/dependency-rule-tried-to-blank-out-primary-key-in-sqlalchemy-when-foreign-key-c